tool/internal/ast/primitives.go

```diff
@@ -275,3 +305,14 @@ func StructLit(typeName string, fields ...*dst.KeyValueExpr) dst.Expr {
 		X:  CompositeLit(Ident(typeName), exprs),
 	}
 }
+
+// CloneTypeParams safely clones a type parameter field list for generic functions.
+// Returns nil if the input is nil.
+func CloneTypeParams(typeParams *dst.FieldList) *dst.FieldList {
+	if typeParams == nil {
+		return nil
+	}
+	cloned, ok := dst.Clone(typeParams).(*dst.FieldList)
+	util.Assert(ok, "typeParams is not a FieldList")
+	return cloned
```

Comment:
And you can use `cloned := util.AssertType[*dst.FieldList](...)` after merging main branch